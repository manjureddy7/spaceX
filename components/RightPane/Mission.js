import React from 'react';
import styles from '../../styles/Mission.module.css';

const Mission = ({missionDetails}) => {
    const {image,missionName,missionIds,launchYear,successfulLaunch,successfulLanding} = missionDetails;

    return(
        <div className={styles.MissionContainer}>
            <div className={styles.ImageContainer}></div>
            <p className={styles.MissionName}>{missionName}</p>
            <p>
                <span className={styles.TextBold}>Mission ids:</span>
                <span className={styles.TextBlue}>{missionIds.length > 0 ? missionIds : ''}</span>
            </p>
            <p>
                <span className={styles.TextBold}>Launch Year: </span>
                <span className={styles.TextBlue}>{launchYear}</span>
            </p>
            <p>
                <span className={styles.TextBold}>Successful Launch: </span>
                <span className={styles.TextBlue}>{successfulLaunch ? 'True': 'False'}</span>
            </p>
            <p>
                <span className={styles.TextBold}>Successful Landing: </span>
                <span className={styles.TextBlue}>{successfulLanding}</span>
            </p>
        </div>
    )
}

export default Mission;