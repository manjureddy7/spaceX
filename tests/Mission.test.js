import React from 'react';
import { render, cleanup, screen } from '@testing-library/react'

import Mission from '../components/RightPane/Mission';

afterEach(cleanup);

const missionDetails = {
    image: '',
    missionName: '',
    missionIds: '',
    launchYear: '',
    successfulLaunch: '',
    successfulLanding: ''
}

describe('Mission Component', () => {
    test('renders Mission component', () => {
      render(<Mission missionDetails={missionDetails} />);
      expect(screen.getByText('Mission ids:')).toBeInTheDocument();
      expect(screen.getByText('Launch Year:')).toBeInTheDocument();
      expect(screen.getByText('Successful Launch:')).toBeInTheDocument();
      expect(screen.getByText('Successful Landing:')).toBeInTheDocument();
      //screen.debug();
    });
    test('Should match snapshot', () => {
        const { asFragment } = render(<Mission missionDetails={missionDetails} />)
        expect(asFragment(<Mission missionDetails={missionDetails} />)).toMatchSnapshot()
    })
});