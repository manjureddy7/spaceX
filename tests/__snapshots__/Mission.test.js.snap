// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mission Component Should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div />
    <p />
    <p>
      <span>
        Mission ids:
      </span>
      <span />
    </p>
    <p>
      <span>
        Launch Year: 
      </span>
      <span />
    </p>
    <p>
      <span>
        Successful Launch: 
      </span>
      <span>
        False
      </span>
    </p>
    <p>
      <span>
        Successful Landing: 
      </span>
      <span />
    </p>
  </div>
</DocumentFragment>
`;
